pyxem/signals/diffraction2d.py

```diff
@@ -718,9 +758,44 @@ def get_direct_beam_position(
                     sig_axes,
                 )
             ]
+
             signal = self.isig[low_x:high_x, low_y:high_y]
-        else:
-            signal = self
+            if show_slice_on_plot:
+                # convert to values, use numpy advanced indexing
+                low_x_, high_x_ = self.axes_manager.signal_axes[0].axis[
+                    (low_x, high_x),
+                ]
+                low_y_, high_y_ = self.axes_manager.signal_axes[1].axis[
+                    (low_y, high_y),
+                ]
+                w, h = high_x_ - low_x_, high_y_ - low_y_
+                marker = hs.plot.markers.Rectangles(
+                    offsets=(low_x_ + w / 2, low_y_ + h / 2),
+                    widths=(w,),
+                    heights=(h,),
+                    color="red",
+                    facecolor="none",
+                )
+                self.add_marker(marker)
+
+        if prefilter_sigma is not None:
+            if not isiterable(prefilter_sigma):
+                # Prefilter in navigation space only
+                prefilter_sigma = (
+                    prefilter_sigma,
+                ) * signal.axes_manager.navigation_dimension + (
+                    0,
+                ) * signal.axes_manager.signal_dimension
+            if self._lazy:
+                from dask_image.ndfilters import gaussian_filter
+
+                gaussian_function = gaussian_filter
+            else:
+                gaussian_function = ndi.gaussian_filter
+
+            signal.filter(
+                gaussian_function, sigma=prefilter_sigma, inplace=True, mode="nearest"
+            )
```

Comment:
@ericpre I think this isn't a bad idea but I've never really had a problem beam centering that needed a gaussian filter.  Do you have a use case for this?

Maybe it would be better to have the "make_linear_plane" be a default instead. That is the computationally cheaper option and it seems to work in almost all cases.

Reply:
> @ericpre I think this isn't a bad idea but I've never really had a problem beam centering that needed a gaussian filter. Do you have a use case for this?

I see some improvement with some dataset acquired on a camera with very limited dynamic range. The advantage of including in the function is to be able to apply the filtering on the sliced signal.

> Maybe it would be better to have the "make_linear_plane" be a default instead. That is the computationally cheaper option and it seems to work in almost all cases.

I have tried `make_linear_plane` and it seems to no play well with have some datasets (other type than the one mentioned just above). I suspect that this is because of the large field of view and the shift is not linear.